drivers/gpio/gpio_tle9104.c

```diff
@@ -0,0 +1,544 @@
+	sys_put_be16(write_frame, buffer_tx);
+	LOG_DBG("writing in register 0x%02X of TLE9104 value 0x%02X, complete frame 0x%04X",
+		write_reg, write_data, write_frame);
+
+	result = spi_transceive_dt(&config->bus, &tx, &rx);
+	if (result != 0) {
+		LOG_ERR("spi_write failed with error %i", result);
+		return result;
+	}
+
+	read_frame = sys_get_be16(buffer_rx);
+	LOG_DBG("received complete frame 0x%04X", read_frame);
+
+	if (!tle9104_check_parity(read_frame)) {
+		LOG_ERR("parity check for received frame of TLE9104 failed");
+		return -EIO;
+	}
+
+	if ((BIT(13) & read_frame) != 0) {
+		LOG_WRN("communication fault reported by TLE9104");
+	}
+
+	*read_reg = FIELD_GET(GENMASK(TLE9104_FRAME_FAULTGLOBAL_POS - 1, TLE9104_FRAME_ADDRESS_POS),
+			      read_frame);
+	*read_data = FIELD_GET(GENMASK(TLE9104_FRAME_ADDRESS_POS - 1, TLE9104_FRAME_DATA_POS),
+			       read_frame);
+
+	return 0;
+}
+
+static int tle9104_write_register(const struct device *dev, enum tle9104_register reg,
+				  uint8_t value)
+{
+	enum tle9104_register read_reg;
+	uint8_t read_data;
+
+	return tle9104_transceive_frame(dev, true, reg, value, &read_reg, &read_data);
+}
+
+static int tle9104_write_state(const struct device *dev)
+{
+	const struct tle9104_config *config = dev->config;
+	struct tle9104_data *data = dev->data;
+	bool spi_update_required = false;
+	uint8_t register_ctrl = 0x00;
+	int result;
+
+	LOG_DBG("writing state 0x%02X to TLE9104", data->state);
+
+	for (size_t i = 0; i < TLE9104_GPIO_COUNT; ++i) {
+		uint8_t mask = GENMASK(i, i);
+		bool current_value = (data->state & mask) != 0;
+		bool previous_value = (data->previous_state & mask) != 0;
+
+		/*
+		 * Setting the OUTx_ON bits results in a high impedance output,
+		 * clearing them pulls the output to ground. Therefore the
+		 * meaning here is intentionally inverted, as this will then turn
+		 * out for a low active open drain output to be pulled to ground
+		 * if set to off.
+		 */
+		if (current_value == 0) {
+			register_ctrl |= TLE9104_CTRL_OUT1ONS_BIT << (2 * i);
+		} else {
+			register_ctrl |= TLE9104_CTRL_OUT1ONC_BIT << (2 * i);
+		}
+
+		if (current_value == previous_value) {
+			continue;
+		}
+
+		if (config->gpio_control[i].port == NULL) {
+			spi_update_required = true;
+			continue;
+		}
+
+		result = gpio_pin_set_dt(&config->gpio_control[i], current_value);
+		if (result != 0) {
+			LOG_ERR("unable to set control GPIO");
+			return result;
+		}
+	}
+
+	if (spi_update_required) {
+		result = tle9104_write_register(dev, TLE9104REGISTER_CTRL, register_ctrl);
+		if (result != 0) {
+			LOG_ERR("unable to set control register");
+			return result;
+		}
+	}
+
+	data->previous_state = data->state;
+
+	return 0;
+}
+
+static int tle9104_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	struct tle9104_data *data = dev->data;
+	int result;
+
+	/* cannot execute a bus operation in an ISR context */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	if (pin > 7) {
```

Comment:
add a define for "7" (shouldn't it be 4? like `TLE9104_GPIO_COUNT`)

Reply:
nice catch, done